cmd/okta-setup.go

```diff
@@ -0,0 +1,93 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/99designs/keyring"
+	"github.com/chanzuckerberg/blessclient/pkg/config"
+	"github.com/chanzuckerberg/blessclient/pkg/util"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/pkg/errors"
+	awsokta "github.com/segmentio/aws-okta/lib"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(oktaSetupCmd)
+}
+
+var oktaSetupCmd = &cobra.Command{
+	Use:           "okta-setup",
+	Short:         "okta-setup sets up Okta login credentials",
+	SilenceErrors: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		expandedConfigFile, err := util.GetConfigPath(cmd)
+		if err != nil {
+			return err
+		}
+		log.Debugf("Reading config from %s", expandedConfigFile)
+		conf, err := config.FromFile(expandedConfigFile)
+		if err != nil {
+			return err
+		}
+		log.Debugf("Parsed config is: %s", spew.Sdump(conf))
+
+		if conf.OktaConfig == nil {
+			return errors.Errorf("The okta_config section is not found in your config")
+		}
+
+		kr, err := awsokta.OpenKeyring(nil)
+		if err != nil {
+			return err
+		}
+		username, err := awsokta.Prompt("Okta username", false)
+		if err != nil {
+			return err
+		}
+
+		password, err := awsokta.Prompt("Okta password", true)
+		if err != nil {
+			return err
+		}
+		fmt.Println()
+
+		organization := conf.OktaConfig.Organization
+		domain := conf.OktaConfig.Domain
+
+		creds := awsokta.OktaCreds{
+			Organization: organization,
+			Username:     username,
+			Password:     password,
+			Domain:       domain,
+		}
+
+		mfaDevice := "phone1"
+		if conf.OktaConfig.MFADevice != nil {
+			mfaDevice = *conf.OktaConfig.MFADevice
+		}
+		if err = creds.Validate(mfaDevice); err != nil {
+			return errors.Wrap(err, "Failed to verify Okta credentials")
+		}
+
+		encoded, err := json.Marshal(creds)
+		if err != nil {
+			return err
+		}
+
+		item := keyring.Item{
+			Key:                         "okta-creds",
```

Comment:
> Is okta-creds the default key for aws-okta?
yep, and if this is set as the same as aws-okta, the same creds can be reused for both tools (for those of us who SSH and use AWS CLI for daily work)

> Should we version contents somehow?
I don't know if it makes sense to, because the credential is not time-based. It's just your Okta password.